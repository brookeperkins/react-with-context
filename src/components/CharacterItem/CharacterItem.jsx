import React from 'react';
import PropTypes from 'prop-types';
import { useTheme } from '../../hooks/themes';


const CharacterItem = ({ name, imageUrl }) => {
  const { theme } = useTheme();

  return <figure className={theme}>
    <img src={imageUrl} alt={name} />
    <figcaption>{name}</figcaption>
  </figure>;
};

CharacterItem.propTypes = {
  name: PropTypes.string.isRequired,
  imageUrl: PropTypes.string.isRequired
};

export default CharacterItem;
