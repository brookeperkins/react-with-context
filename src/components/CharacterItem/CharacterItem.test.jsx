import React from 'react';
import { render, cleanup } from '@testing-library/react';
import CharacterItem from './CharacterItem';

describe('<CharacterItem />', () => {
  afterEach(() => cleanup());

  it('renders single character', () => {
    const { asFragment } = render(<CharacterItem
      name="Arnold"
      imageUrl="arnold.jpg"
    />);
    expect(asFragment()).toMatchSnapshot();
  });
});
