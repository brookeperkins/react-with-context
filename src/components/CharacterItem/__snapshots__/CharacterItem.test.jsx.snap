// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CharacterItem /> renders single character 1`] = `
<DocumentFragment>
  <figure>
    <img
      alt="Arnold"
      src="arnold.jpg"
    />
    <figcaption>
      Arnold
    </figcaption>
  </figure>
</DocumentFragment>
`;

exports[`CharacterItem component renders single character 1`] = `
<DocumentFragment>
  <figure>
    <img
      alt="Arnold"
      src="arnold.jpg"
    />
    <figcaption>
      Arnold
    </figcaption>
  </figure>
</DocumentFragment>
`;
